lib/new_relic/agent/new_relic_service.rb

```diff
@@ -362,8 +366,8 @@ def create_and_start_http_connection
         conn = create_http_connection
         start_connection(conn)
         conn
-      rescue Timeout::Error
-        ::NewRelic::Agent.logger.info('Timeout while attempting to connect. You may need to install system-level CA Certificates, as the ruby agent no longer includes these.')
+      rescue Net::OpenTimeout
```

Comment:
Is this the only type of timeout that needs to be rescued here? I'm curious about the other types of timeouts

Reply:
Yep, but great question here. Net::HTTP will only throw 1 of 3 exceptions: an open timeout, a read timeout, or a write timeout (and that last one wasn't added until Ruby 2.6). The part that confused me was the ability to specify an "ssl timeout" without having a corresponding ssl timeout error to worry about.

We rescue all 3 error types elsewhere, but in this exact context we are only opening the connection and not reading or writing across it, so just the open timeout is used.

Reply:
I've decided it's better to leave SSL timeouts up to OpenSSL (which we engage separately elsewhere in this class with code untouched by the PR) and I removed the Net::HTTP specific `ssl_timeout` value.

Reply:
That makes sense, thanks for explaining why this one was just the open timeout!